### PR TITLE
Merge some bugfixes for DIANN dataset into Development

### PR DESCRIPTION
**Pull request by swaraj-neu**
- Default the DIANN 2.0+ checkbox to unchecked
- Auto detect the DIANN dataset file type and automatically toggle the checkbox for better user experience, still retaining the ability to let user manually toggle the checkbox if required
- Show a warning if the user selects an invalid state of the checkbox based on the type of file uploaded
- Handle application crashing in case if the user uses an invalid configuration while uplaoding the data
- Write test cases for all edge cases of the DIANN file version detection

## Motivation and Context

The DIANN dataset support in MSstatsShiny previously defaulted the DIANN 2.0+ checkbox to an unsafe state, lacked robust auto-detection of DIANN file versions, and could crash or mis-handle uploads when user selection conflicted with file contents. This PR improves UX and robustness by auto-detecting DIANN file format from a file preview, keeping the user able to override the choice, warning on mismatches, and preventing crashes from invalid upload configurations.

## Solution Summary

On file preview the app now infers DIANN 1.x vs DIANN 2.0+ using column-name patterns and updates the DIANN 2.0+ checkbox to match detected format (while allowing manual override). A mismatch watcher warns the user if their manual selection conflicts with detection. File-preview and upload flows use a centralized, resilient preview reader and added error handling to avoid application crashes. Unit tests were added to cover detection and preview edge cases, including Parquet schema-only previews.

## Detailed Changes

- NAMESPACE
  - No changes to NAMESPACE in this PR (imports remain as-is).

- UI (R/module-loadpage-ui.R)
  - DIANN 2.0+ checkbox default changed from TRUE to FALSE so the pre-2.0 intensity input panel is shown by default when DIANN is selected.

- Server (R/module-loadpage-server.R)
  - Centralized preview reads via .read_preview() for preview-based UI features (applies to DIANN as well as Metamorpheus PTM).
  - preview_data reactive introduced to hold the preview result.
  - Added last_detected_diann_format reactive to avoid redundant notifications.
  - After preview, run .is_diann_2plus(preview) to detect DIANN file version.
  - Auto-update DIANN 2.0+ checkbox via updateCheckboxInput() only when detected state changes; show informative notifications for detected DIANN 2.0+ vs 1.x.
  - observeEvent on user changes to diann_2plus emits a warning if manual selection conflicts with detected format.
  - get_data / upload flow hardened: getData(input) errors are caught, modal spinner removed, a failure notification is shown, and NULL is returned to prevent crashes.
  - Preview failures produce standardized "Could not preview file. Please verify the file format." warning.

- Utilities (R/utils.R)
  - Added .read_preview(filepath, filename = NULL, nrows = 100):
    - Reads up to nrows for delimited text using data.table::fread.
    - For Parquet (.parquet/.pq), uses Arrow to open dataset schema and returns an empty data.frame with correct column names (schema-only), avoiding full reads.
    - Returns NULL on read errors.
  - Added .is_diann_2plus(preview_df):
    - Detects DIANN 2.0+ by presence of numbered fragment quantity columns matching "^Fr\\.[0-9]+\\.Quantity$" and absence of legacy fragment columns ("Fragment.Quant.Corrected", "FragmentQuantCorrected").
    - Returns FALSE for NULL or empty inputs or when legacy columns are present.

- Other server/UI behaviors
  - Metamorpheus PTM preview logic generalized to use .read_preview and standardized warning on preview failure.
  - extract/resolve helper functions present/used for PTM modification ID UI.
  - Various UI renderUI outputs continue to rely on preview_data where applicable.

## Unit Tests Added / Modified

- tests/testthat/test-utils.R
  - New tests for .is_diann_2plus:
    - TRUE for DIANN 2.0+ patterns (Fr.0.Quantity … Fr.11.Quantity).
    - FALSE for DIANN 1.x legacy columns (Fragment.Quant.Corrected, FragmentQuantCorrected).
    - Precedence: legacy fragment columns cause FALSE when both legacy and numbered columns coexist.
    - Defensive cases: NULL preview, empty data.frame, data without fragment columns.
  - New tests for .read_preview:
    - CSV/TSV parsing with nrows limiting.
    - Returns NULL for missing/non-existent files.
    - Tolerant handling for malformed bytes.
    - Behavior when filename is NULL.
    - Parquet (.parquet and .pq) schema-only reads via Arrow (returns empty data.frame with correct columns).
  - Integration tests:
    - End-to-end checks reading temporary CSV/parquet preview and verifying .is_diann_2plus classification for DIANN 1.x vs 2.0+.

## Coding Guidelines

- No coding guideline violations detected:
  - Internal helpers use dot-prefix naming (.read_preview, .is_diann_2plus).
  - Non-exported internals are annotated with @noRd where present.
  - Error handling uses tryCatch to return NULL on preview errors for graceful degradation.
  - Shiny reactive idioms (reactiveVal, observe, observeEvent, updateCheckboxInput) are used appropriately.
  - Tests are organized and cover edge cases for detection and preview behavior.

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -38,7 +38,6 @@ importFrom(DT,renderDT)
 importFrom(DT,renderDataTable)
 importFrom(Hmisc,describe)
 importFrom(MSstatsBioNet,annotateProteinInfoFromIndra)
-importFrom(MSstatsBioNet,deleteEdgeFromNetwork)
```
Comment (tonywu1999) [inline NAMESPACE:42]: Rebase to devel and install the latest version of MSstatsBioNet from Bioconductor devel / github.  Then rerun devtools::document().  This shouldn't be deleted.
